File: services/catalog-service/src/main/java/com/tap2eat/catalog/dtos/response/internal/ValidatedOrderItemResponse.java
package com.tap2eat.catalog.dtos.response.internal;

import java.math.BigDecimal;
import java.util.List;

public record ValidatedOrderItemResponse(
        String productId,
        String productName,
        Integer quantity,
        BigDecimal unitPrice,
        List<ValidatedModifierResponse> selectedModifiers,
        BigDecimal subtotal
) {
}

File: services/order-service/Integrations/Catalog/Dtos/ValidatedOrderItemResponse.cs
namespace OrderService.Integrations.Catalog.Dtos;

public sealed class ValidatedOrderItemResponse
{
    public string ProductId { get; set; } = string.Empty;

    public string ProductName { get; set; } = string.Empty;

    public int Quantity { get; set; }

    public decimal UnitPrice { get; set; }

    public List<ValidatedModifierResponse> SelectedModifiers { get; set; } = [];

    public decimal Subtotal { get; set; }
}
